### PR TITLE
Use the clang crate to autodetect what postgres version we're using

### PR DESCRIPTION
**Pull request by stillinbeta**
No more trying to manually specify --features!

### CODE REVIEW

File: pg-extend/Cargo.toml [hunk 1]
```diff
@@ -26,4 +26,5 @@ postgres-11 = []
 [dependencies]
 
 [build-dependencies]
-bindgen = "0.46"
+bindgen = "0.41"
```
Comment (stillinbeta) [inline pg-extend/Cargo.toml]: bindgen 0.42+ requires clang-sys 0.26. clang requires clang-sys 0.23. This downgrade doesn't appear to break anything and solves this dep conflict.

File: pg-extend/pg_majorversion.h [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#include <pg_config.h>
+
+char pg_majorversion[2] = PG_MAJORVERSION;
```
Comment (stillinbeta) [inline pg-extend/pg_majorversion.h]: clang doesn't like parsing macros (that I can tell). But this makes it easy to pick out the version

File: pg-extend/Cargo.toml (hunk 1 above)
Comment (stillinbeta) [reply pg-extend/Cargo.toml]: ...https://github.com/rust-lang/rust-bindgen/pull/1410 ah maybe this isn't an option. hmm.

**Comment by stillinbeta** [comment]
@bluejekyll unfortunately it looks like we need a newer bindgen to fix that. I'm going to bug the authors of `clang` about releasing a  newer version that tags 0.26

**Comment by stillinbeta** [comment]
@bluejekyll good to go now!

**Comment by bluejekyll** [comment]
Thanks for the PR!

**Pull request merged**